Ble tx improvements

iNetro opened the pull request:
Git log still looks big. Not sure how to fix this @gullradriel T_T. Anyway added improvements to BLETx

NotherNgineer commented:
Regarding the huge list of commits:
I'm not very skilled at git so I don't know if this will help, but here's what I do, just for comparison:

1.  I always keep my "next" branch in sync with the eried/next branch.  E.g. On the github website, at some point after PR's are committed to eried/next, I view my "next" branch and click on "sync fork" to update my "next" branch to match eried/next.  Note that my "next" branch never gets AHEAD of eried/next (I never sync my development branches to my "next" branch).

2.  I use the "git" command line on my linux system to pull down a copy of my "next" branch as a starting point for code changes.  E.g. "sudo git clone --recurse-submodules https://github.com/NotherNgineer/portapack-mayhem.git", then "chown -R mark:mark *".  Then I edit the code & test on my local system, keeping track of which files I changed.

3.  Only when I'm ready to submit a PR, I create a new development branch from my "next" branch using github.com.  To commit my changes, I open my development branch on github.com, browse to the folder where the files are located, and use the "upload" option on github.com to drag & drop my updated files to the appropriate folder(s) in my development branch.  (I don't use "git push" commands mainly because I don't know what I'm doing.)

4.  To make subsequent fixes to the PR (i.e. to address reviewer feedback or CLang issues), I drag updated files to the same folder(s) in my development branch same as mentioned above.  (These all show up as new commits.)

5.  Also note that I don't enable any git awareness in the VS Code editor either (thus it doesn't show up as a change in git every time I save local changes to a file).

6.  At times in that past I have managed to goof up my "next" branch, in which case I've created a pristine new branch from eried/next, renamed my "next" branch to "old-next", renamed my new branch to "next", then deleted the "old-next" branch.

Maybe other developers have their own (perhaps better) way of doing things but just FYI.

iNetro commented:
> Regarding the huge list of commits:
> I'm not very skilled at git so I don't know if this will help, but here's what I do, just for comparison:
> 
> 1.  I always keep my "next" branch in sync with the eried/next branch.  E.g. On the github website, at some point after PR's are committed to eried/next, I view my "next" branch and click on "sync fork" to update my "next" branch to match eried/next.  Note that my "next" branch never gets AHEAD of eried/next (I never sync my development branches to my "next" branch).
> 
> 2.  I use the "git" command line on my linux system to pull down a copy of my "next" branch as a starting point for code changes.  E.g. "sudo git clone --recurse-submodules https://github.com/NotherNgineer/portapack-mayhem.git", then "chown -R mark:mark *".  Then I edit the code & test on my local system, keeping track of which files I changed.
> 
> 3.  Only when I'm ready to submit a PR, I create a new development branch from my "next" branch using github.com.  To commit my changes, I open my development branch on github.com, browse to the folder where the files are located, and use the "upload" option on github.com to drag & drop my updated files to the appropriate folder(s) in my development branch.  (I don't use "git push" commands mainly because I don't know what I'm doing.)
> 
> 4.  To make subsequent fixes to the PR (i.e. to address reviewer feedback or CLang issues), I drag updated files to the same folder(s) in my development branch same as mentioned above.  (These all show up as new commits.)
> 
> 5.  Also note that I don't enable any git awareness in the VS Code editor either (thus it doesn't show up as a change in git every time I save local changes to a file).
> 
> 6.  At times in that past I have managed to goof up my "next" branch, in which case I've created a pristine new branch from eried/next, renamed my "next" branch to "old-next", renamed my new branch to "next", then deleted the "old-next" branch.
> 
> Maybe other developers have their own (perhaps better) way of doing things but just FYI.

This is helpful. FWIW, I use GitKraken which may be messing things up.

iNetro commented:
> > Regarding the huge list of commits:
> > I'm not very skilled at git so I don't know if this will help, but here's what I do, just for comparison:
> > 
> > 1.  I always keep my "next" branch in sync with the eried/next branch.  E.g. On the github website, at some point after PR's are committed to eried/next, I view my "next" branch and click on "sync fork" to update my "next" branch to match eried/next.  Note that my "next" branch never gets AHEAD of eried/next (I never sync my development branches to my "next" branch).
> > 
> > 2.  I use the "git" command line on my linux system to pull down a copy of my "next" branch as a starting point for code changes.  E.g. "sudo git clone --recurse-submodules https://github.com/NotherNgineer/portapack-mayhem.git", then "chown -R mark:mark *".  Then I edit the code & test on my local system, keeping track of which files I changed.
> > 
> > 3.  Only when I'm ready to submit a PR, I create a new development branch from my "next" branch using github.com.  To commit my changes, I open my development branch on github.com, browse to the folder where the files are located, and use the "upload" option on github.com to drag & drop my updated files to the appropriate folder(s) in my development branch.  (I don't use "git push" commands mainly because I don't know what I'm doing.)
> > 
> > 4.  To make subsequent fixes to the PR (i.e. to address reviewer feedback or CLang issues), I drag updated files to the same folder(s) in my development branch same as mentioned above.  (These all show up as new commits.)
> > 
> > 5.  Also note that I don't enable any git awareness in the VS Code editor either (thus it doesn't show up as a change in git every time I save local changes to a file).
> > 
> > 6.  At times in that past I have managed to goof up my "next" branch, in which case I've created a pristine new branch from eried/next, renamed my "next" branch to "old-next", renamed my new branch to "next", then deleted the "old-next" branch.
> > 
> > Maybe other developers have their own (perhaps better) way of doing things but just FYI.
> 
> This is helpful. FWIW, I use GitKraken which may be messing things up.

After this I should probably do your last point and redo a next branch and delete the old one. Maybe something just got messed up.

gullradriel reviewed (approved):
Good to go

gullradriel commented:
Yes, it would be better to start with a fresh branch and just use git command without bloat around x-D
